enterprise/server/scheduling/task_router/task_router.go

```diff
@@ -212,11 +253,17 @@ type Router interface {
 	// false otherwise. Note: Applies() must be deterministic.
 	Applies(params routingParams) bool
 
-	// Returns the routing info (preferredNodeLimit and routingKey) for the
-	// provided routing parameters. The preferredNodeLimit is the number of
-	// preferred executor nodes that should be used, and the routing key is the
-	// Redis key where the list of preferred executor nodes are stored.
-	RoutingInfo(params routingParams) (int, string, error)
+	// Returns the routing info (preferredNodeLimit and routingKeys) for the
+	// provided routing parameters.
+	//
+	// The preferredNodeLimit is the number of preferred executor nodes that
+	// should be used.
+	//
+	// The routing keys are the Redis keys where the list of preferred executor
+	// nodes are stored. When marking a task complete, only the first key is
+	// used to update the node list. When ranking nodes, keys should be treated
+	// in order of most preferred to least preferred.
+	RoutingInfo(params routingParams) (int, []string, error)
```

Comment:
can we start with a single routingKey and see if there's an issue?

Reply:
Done